accumulator/accumulate_signals_workflow.go

```
@@ -0,0 +1,168 @@
+		return signalToSignalTimeout, nil
+	}
+	return totalLeft, nil
+}
+
+// AccumulateSignalsWorkflow workflow definition
+func AccumulateSignalsWorkflow(ctx workflow.Context, greetings GreetingsInfo) (allGreetings string, err error) {
+	log := workflow.GetLogger(ctx)
+	var a AccumulateGreeting
+	if greetings.GreetingsList == nil {
+		greetings.GreetingsList = []AccumulateGreeting{}
+	}
+	if greetings.UniqueGreetingKeys == nil {
+		greetings.UniqueGreetingKeys = make(map[string]bool)
+	}
+	var unprocessedGreetings []AccumulateGreeting
+	if greetings.startTime.IsZero() {
+		greetings.startTime = workflow.Now(ctx)
+	}
+	exitRequested := false
+
+	ao := workflow.ActivityOptions{
+		StartToCloseTimeout: 100 * time.Second,
+	}
+	ctx = workflow.WithActivityOptions(ctx, ao)
+
+	for !workflow.GetInfo(ctx).GetContinueAsNewSuggested() {
+
+		timeout, err := a.GetNextTimeout(ctx, greetings.startTime, exitRequested)
+		childCtx, cancelHandler := workflow.WithCancel(ctx)
+		selector := workflow.NewSelector(ctx)
+
+		if err != nil {
+			log.Error("Error calculating timeout")
+			return "", err
+		}
+		log.Debug("Awaiting for " + timeout.String())
+		selector.AddReceive(workflow.GetSignalChannel(ctx, "greeting"), func(c workflow.ReceiveChannel, more bool) {
+			c.Receive(ctx, &a)
+			unprocessedGreetings = append(unprocessedGreetings, a)
+			log.Debug("Signal Received with text: " + a.GreetingText + ", more?: " + strconv.FormatBool(more) + "\n")
+			cancelHandler() // cancel timer future
+			a = AccumulateGreeting{}
+		})
+		selector.AddReceive(workflow.GetSignalChannel(ctx, "exit"), func(c workflow.ReceiveChannel, more bool) {
+			c.Receive(ctx, nil)
+			exitRequested = true
+			cancelHandler() // cancel timer future
+			log.Debug("Exit Signal Received, more?: " + strconv.FormatBool(more) + "\n")
+		})
+
+		timerFuture := workflow.NewTimer(childCtx, timeout)
+		selector.AddFuture(timerFuture, func(f workflow.Future) {
+			log.Debug("Timer fired \n")
```

This can be a workflow cancel error, it should not be swallowed